compiler/src/iree/compiler/Codegen/Common/GPU/WorkgroupReordering.cpp

```diff
@@ -153,7 +269,11 @@ static LogicalResult reorderWorkgroupsInFunc(FunctionOpInterface funcOp,
   if (strategy == ReorderWorkgroupsStrategy::Swizzle) {
     std::tie(newWorkgroupIdX, newWorkgroupIdY) =
         makeSwizzledIds(funcOp.getLoc(), builder, workgroupIdX, workgroupIdY,
-                        workgroupCntX, workgroupCntY, swizzleTile);
+                        workgroupCntX, workgroupCntY, reorderWgTileSize);
+  } else if (strategy == ReorderWorkgroupsStrategy::ChipletGroup) {
```

Comment:
Add a check here that returns failure if `numXCDs <= 1` to reflect the assert in `makeChipletGroupedIds`

Reply:
MakeChipletGroupedIds is designed exclusively for the chiplet group strategy; so, it seems that an assertion for the swizzle method is unnecessary.

Reply:
I'm mainly asking because there is an assert for it in `makeChipletGroupedIds`, meaning if the assert gets stripped and someone calls this function with the wrong number of xcds somewhere, then it's UB. Asserts should only be used to catch inconsistencies internal to the compiler.